src/instructlab/configuration.py

```diff
@@ -586,13 +793,209 @@ def ensure_storage_directories_exist():
+        nproc_per_node=4,
+        effective_batch_size=96,
+        lora_quantize_dtype=None,
+        lora_rank=0,
+        max_batch_len=60000,
+        max_seq_len=4096,
+        save_samples=1000,
+        deepspeed_cpu_offload_optimizer=True,
+        is_padding_free=True,
+        model_path="~/.cache/instructlab/models/granite-7b-starter",
+    )
+
+    EIGHT_L_FORTY_GPU = _train(
+        additional_args={
+            "warmup_steps": 10,
+            "learning_rate": 2e-6,
+            "lora_dropout": 0.1,
+            "lora_alpha": 32,
+        },
+        ckpt_output_dir="~/.local/share/instructlab/checkpoints",
+        data_output_dir="~/.local/share/instructlab/internal",
+        data_path="~/.local/share/instructlab/datasets",
+        nproc_per_node=4,
+        effective_batch_size=96,
+        lora_quantize_dtype=None,
+        lora_rank=0,
+        max_batch_len=30000,
+        max_seq_len=4096,
+        save_samples=1000,
+        deepspeed_cpu_offload_optimizer=False,
+        is_padding_free=True,
+        model_path="~/.cache/instructlab/models/granite-7b-starter",
+    )
+
+    FOUR_L_FORTY_GPU = _train(
+        additional_args={
+            "warmup_steps": 10,
+            "learning_rate": 2e-6,
+            "lora_dropout": 0.1,
+            "lora_alpha": 32,
+            "deepspeed_cpu_offload_optimizer_ratio": 1,
+            "deepspeed_cpu_offload_optimizer_pin_memory": False,
+        },
+        ckpt_output_dir="~/.local/share/instructlab/checkpoints",
+        data_output_dir="~/.local/share/instructlab/internal",
+        data_path="~/.local/share/instructlab/datasets",
+        nproc_per_node=4,
+        effective_batch_size=96,
+        lora_quantize_dtype=None,
+        lora_rank=0,
+        max_batch_len=30000,
+        max_seq_len=4096,
+        save_samples=1000,
+        deepspeed_cpu_offload_optimizer=True,
+        is_padding_free=True,
+        model_path="~/.cache/instructlab/models/granite-7b-starter",
+    )
+
+    EIGHT_L_FOUR_GPU = _train(
+        additional_args={
+            "warmup_steps": 10,
+            "learning_rate": 2e-6,
+            "lora_dropout": 0.1,
+            "lora_alpha": 32,
+            "deepspeed_cpu_offload_optimizer_ratio": 1,
+            "deepspeed_cpu_offload_optimizer_pin_memory": False,
+        },
+        ckpt_output_dir="~/.local/share/instructlab/checkpoints",
+        data_output_dir="~/.local/share/instructlab/internal",
+        data_path="~/.local/share/instructlab/datasets",
+        nproc_per_node=4,
+        effective_batch_size=96,
+        lora_quantize_dtype=None,
+        lora_rank=0,
+        max_batch_len=30000,
+        max_seq_len=4096,
+        save_samples=1000,
+        deepspeed_cpu_offload_optimizer=True,
+        is_padding_free=True,
+        model_path="~/.cache/instructlab/models/granite-7b-starter",
+    )
+
     # create exper_args file for users to see/edit
     if not os.path.isfile(DEFAULTS.TRAIN_ADDITIONAL_OPTIONS_FILE):
```

Comment:
Is there any way we can avoid storing these large config blocks in the configuration code and use a set of config files instead? a middle-ground is having Python files for each of these config objects if moving some .yaml files around is a problem, but having data stored in code is kinda kludgey.